### PR TITLE
Revamp rojak

### PR DESCRIPTION
**Pull request by samueljklee**
As mentioned offline
- revamp rojak to have 2 ways of searching for dishes
   - use an existing dataset to provide title, ingredients, and instructions. (this PR)
   - use ML to generate title, ingredient, instruction based on the input. (future)

### changes in this pr
- change dataset, so we don't have to recreate our own url to share files
- use existing recipe dataset as a knowledge graph to find
- based on ingredients of each recipe, find keywords to use it to build nodes in the graph
- use stemming method to break down words to their simpler form
   - example of stemming
```
 'halves': 'halv',
 'tablespoons': 'tablespoon',
 'butter': 'butter',
 'ounce': 'ounc',
 'cans': 'can',
 'condensed': 'condens',
 'finely': 'fine',
 'diced': 'dice',
 'packages': 'packag',
 'refrigerated': 'refriger',
```
- doesn't have to make sense, it's just a way to break down input queries and try to match KG.
- simple ranking implemented
   - for each node that exists, add a point

Example usage (currently set top 10 scores to output):
image

Please review and criticize this long due update @binhonglee

### CODE REVIEW

File: foodRequest.py [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
+handler.setFormatter(formatter)
+root.addHandler(handler)
+# Logging Ends
+
 @app.route('/todo/api/v1.0/recipes/', methods=['POST'])
 @cross_origin()
 def get_data():
     if "content" in request.json:
-        name = json.loads(request.json)["content"]
+        query = request.json["content"]
```
Comment (binhonglee) [inline foodRequest.py:30]: Noob question, what's the difference between using and not using `json.loads()`?

File: RojakCore.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        for _, _, node in top_x_results:
+            res.append({
+                "title": node.title,
+                "ingredients": node.ingredients,
+                "instructions": node.instructions.split('\n')
+            })
+        return res
+
+    def search(self, query: str):
+        query = "Chicken Pie with Biscuit Crust"
```
Comment (binhonglee) [inline RojakCore.py]: Might want to remove this?

File: RojakCore.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        for tag in tags:
+            KG[tag].add(recipe)
+    logging.debug("Number of tags: {}".format(len(KG)))
+    return KG
+
+def create_dataframe(list_of_json_dish: List[str]) -> pd.DataFrame:
+    # pandas convert to dataframe
+    df = pd.read_json(json.dumps(list_of_json_dish))
+    logging.debug("Number of recipes: {}".format(len(df)))
+    # df.dropna(inplace=True)
```
Comment (binhonglee) [inline RojakCore.py]: Remove this too?

**Review by binhonglee — APPROVED** [review]
Everything else LGTM

File: foodRequest.py (hunk 1 above)
Comment (samueljklee) [reply foodRequest.py:30]: `request.json` returns `json` format so we don't have to load it with `json` library again.

**Pull request merged**